python/lsst/ip/isr/calibType.py

```diff
@@ -0,0 +1,600 @@
+        self._metadata[self._OBSTYPE + "_VERSION"] = self._VERSION
+
+    def updateMetadata(self, setDate=False, **kwargs):
+        """Update metadata keywords with new values.
+
+        Parameters
+        ----------
+        setDate : `bool`, optional
+            Ensure the metadata CALIBDATE fields are set to the current datetime.
+        kwargs :
+            Set of key=value pairs to assign to the metadata.
+        """
+        mdOriginal = self.getMetadata()
+        mdSupplemental = dict()
+
+        self._metadata["DETECTOR"] = self._detectorName
+        self._metadata["DETECTOR_SERIAL"] = self._detectorSerial
+
+        if setDate:
+            date = datetime.datetime.now()
+            mdSupplemental['CALIBDATE'] = date.isoformat()
+            mdSupplemental['CALIB_CREATION_DATE'] = date.date().isoformat()
+            mdSupplemental['CALIB_CREATION_TIME'] = date.time().isoformat()
+
+        mdSupplemental.update(kwargs)
+        mdOriginal.update(mdSupplemental)
+
+    @classmethod
+    def readText(cls, filename):
+        """Read calibration representation from a yaml/ecsv file.
+
+        Parameters
+        ----------
+        filename : `str`
+            Name of the file containing the calibration definition.
+
+        Returns
+        -------
+        calib : `~lsst.ip.isr.IsrCalibType`
+            Calibration class.
+
+        Raises
+        ------
+        RuntimeError :
+            Raised if the filename does not end in ".ecsv" or ".yaml".
+        """
+        if filename.endswith((".ecsv", ".ECSV")):
+            data = Table.read(filename, format='ascii.ecsv')
+            return cls.fromTable([data])
+
+        elif filename.endswith((".yaml", ".YAML")):
+            with open(filename, 'r') as f:
+                data = yaml.load(f, Loader=yaml.CLoader)
+            return cls.fromDict(data)
+        else:
+            raise RuntimeError(f"Unknown filename extension: {filename}")
+
+    def writeText(self, filename, format='auto'):
+        """Write the calibration data to a text file.
+
+        Parameters
+        ----------
+        filename : `str`
+            Name of the file to write.
+        format : `str`
+            Format to write the file as.  Supported values are:
+                ``"auto"`` : Determine filetype from filename.
+                ``"yaml"`` : Write as yaml.
+                ``"ecsv"`` : Write as ecsv.
+        Returns
+        -------
+        used : `str`
+            The name of the file used to write the data.
+
+        Raises
+        ------
+        RuntimeError :
+            Raised if filename does not end in a known extension, or
+            if all information cannot be written.
+
+        Notes
+        -----
+        The file is written to YAML/ECSV format and will include any
+        associated metadata.
+
+        """
+        if format == 'yaml' or (format == 'auto' and filename.lower().endswith((".yaml", ".YAML"))):
```

Comment:
I think this means that I can pass in a file name of "hello.ecsv" and say "format=yaml" and end up with a yaml file that has a .ecsv extension.  In Defects I force the extension to be correct and I think you have to do that here -- it's why the returned filename doesn't have to match the supplied filename.